lib/i18n-transform.js

```diff
@@ -52,93 +52,172 @@ var _getLanguageFromI18n = function(i18nObj) {
         return language;
     },
 
-    _extractPreferredLanguage = function(r, i18nArray, languages){
+    _isLanguageMatch = function (i18n, language) {
+        var i18nLanguage = _getLanguageFromI18n(i18n);
+
+        return language.code.toUpperCase() === i18nLanguage.code.toUpperCase() &&
```

Comment:
maybe split that into few lines, this is confusing :) especially with all parentheses